File: configlibrary/src/main/java/com/shmedo/configlibrary/ble/cmd/entity/WorkModeEntity.java
package com.shmedo.configlibrary.ble.cmd.entity;

import com.shmedo.configlibrary.ble.exception.DASParameterException;
import com.shmedo.configlibrary.ble.interfaces.Validater;

/**
 * 项目名：  mCloudapp <br/>
 * 包名：    com.shmedo.core.cmd.entity <br/>
 * 创建者:   gonghe <br/>
 * 创建时间:  2020/4/23 <br/>
 * 描述：    设备工作模式参数
 */
public class WorkModeEntity implements Validater {
    private int model;

    public WorkModeEntity(int model) {
        this.model = model;
    }

    @Override
    public void validate() {
        if (model != 0 && model != 1 && model != 2 && model != 3)
            throw new DASParameterException("工作模式错误");
    }

    @Override
    public String toString() {
        return String.valueOf(this.model);
    }
}

File: lib_device_base/src/main/java/com/shmedo/lib/device/base/iot_cmd/entity/WorkModeEntity.kt
package com.shmedo.lib.device.base.iot_cmd.entity

import com.shmedo.lib.device.base.Validater

/**
 * 创建者:   gonghe <br></br>
 * 创建时间:  1/10/21 <br></br>
 * 描述：       生成ADME 工作模式参数拼接指令
 */
class WorkModeEntity(
    var mode //工作模式(1:正常工作模式，2:低功耗模式)
    : String = "1"
) : Validater {
    override fun validate() {}

    override fun toString(): String {
        return "mode=$mode"
    }
}